Assets/Scripts/DetectLookedAtInteractive.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+
+/// <summary>
+/// Detects interactive elements the player is looking at.
+/// </summary>
+public class DetectLookedAtInteractive : MonoBehaviour
+{
+    [Tooltip("Starting point of raycast used to detect interactives.")]
+    [SerializeField]
+    private Transform raycastOrigin;
+
+    [Tooltip("How far from the raycastOrigin we will search for interactive elements.")]
+    [SerializeField]
+    private float maxRange = 5.0f;
+
+    /// <summary>
+    /// Event raised when the player looks at a different IInteractive.
+    /// </summary>
+    public static event Action<IInteractive> LookedAtInteractiveChanged;
+
+    public IInteractive LookedAtInteractive
+    {
+        get { return lookedAtInteractive; }
+        private set
+        {
+            bool isInteractiveChanged = value != lookedAtInteractive;
+            if (isInteractiveChanged)
+            {
+                lookedAtInteractive = value;
+                LookedAtInteractiveChanged?.Invoke(lookedAtInteractive);
+            }
+            
```

Comment:
Remove extra lines / whitespace unless you are systematically adding extra whitespace / line breaks to enhance the readability of your document (for example, by having a blank line separating distinct elements).